Documentation/Diagnostics/PH2147.md

```diff
@@ -12,19 +12,50 @@
 
 ## Introduction
 
-This rule flags variables (local variables, foreach variables, for loop variables, using variables, and out parameters) that are named exactly `_` (single underscore). This rule does not flag proper C# discards (both anonymous discards `_` and typed discards like `string _`).
+This rule flags:
+1. **Unnecessary typed discards**: Typed discards like `out string _` when anonymous discard `out _` would work equally well
+2. **Variables named exactly `_`**: Local variables, foreach variables, for loop variables, using variables, and out parameters that are named exactly `_` (single underscore)
+
+This rule does not flag proper C# anonymous discards (`_`) or typed discards that are necessary for overload resolution.
 
 ## Reason
 
-Variables named exactly `_` can be confused with C# discards. This can lead to unintended behavior where developers think they are using a discard but are actually creating a variable that captures the value.
+**For unnecessary typed discards**: Anonymous discards have clearer intent and are more concise than typed discards when the type information is not needed for overload resolution.
+
+**For variables named `_`**: Variables named exactly `_` can be confused with C# discards. This can lead to unintended behavior where developers think they are using a discard but are actually creating a variable that captures the value.
 
 ## How to fix violations
 
-Either use a proper discard (if you don't need the value) or use a more descriptive variable name (if you do need the value).
+**For unnecessary typed discards**: Replace typed discards with anonymous discards when the type is not needed for overload resolution.
```

Comment:
The fact that we have different explanation in the documentation, gives me the hint we should use 2 separate PH codes for these.

Reply:
Oh, in fact we have. Please split the documentation file into the 2 PH numbers.

Reply:
And, btw, this is not for PH2147, but in fact PH2153 ans PH2154